hello.py

```diff
@@ -386,180 +399,460 @@ def image_ref(self, repo):
+        # args = '-i'
+        entry = '--entrypoint=\'["/bin/sh", "-c"]\''
+        for a,b in BenchRunner.CMD_STDIN[repo].mount:
+            a = os.path.join(os.path.dirname(os.path.abspath(__file__)), a)
+            a = tmp_copy(a)
+            args += ' --mount type=bind,src=%s,dst=%s,options=rbind' % (a,b)
+        
+        cmd = ('%s record --period=%s --file=%s/%s.txt %s %s %s/%s:latest' %
+            (optimize_bin, period, accessed_files_dir, repo , entry, args, target, repo))
+        
+        if BenchRunner.CMD_STDIN[repo].stdin_sh:
+            cmd = '%s -- %s' % (cmd, BenchRunner.CMD_STDIN[repo].stdin_sh) # e.g., sh -c
+        
+        return cmd
+
+    def optimize_cmd_arg_wait_cmd(self, repo, optimize_bin, accessed_files_dir, target, period):
+
+        entry = '--entrypoint=\'["/bin/sh", "-c"]\''
+        args = ""
+        if BenchRunner.CMD_ARG_WAIT[repo].arg != "":
+            args = args + '--args=\'["%s"]\'' % BenchRunner.CMD_ARG_WAIT[repo].arg.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        if len(BenchRunner.CMD_ARG_WAIT[repo].env) > 0:
+            env = " ".join(["-e %s=%s" % (k, v) for k, v in BenchRunner.CMD_ARG_WAIT[repo].env.items()])
+            args = args + ' --env=\'["%s"]\'' % env.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        
+        # if BenchRunner.CMD_ARG_WAIT[repo].waitline != "":
+        #     optimize_bin = "%s --wait-on-line=\"%s\"" % (optimize_bin, BenchRunner.CMD_ARG_WAIT[repo].waitline)
+        
+        cmd = ('%s record --period=%s --file=%s/%s.txt %s %s %s/%s:latest' %
+            (optimize_bin, period, accessed_files_dir, repo , entry, args, target, repo))
+        
+        return cmd
+    
+    def optimize_cmd_url_wait_cmd(self, repo, optimize_bin, accessed_files_dir, target, period):
+        entry = '-entrypoint \'["/bin/sh", "-c"]\''
+        args = ""
+        if BenchRunner.CMD_URL_WAIT[repo].arg != "":
+            args = args + '--args=\'["%s"]\'' % BenchRunner.CMD_URL_WAIT[repo].arg.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        if len(BenchRunner.CMD_URL_WAIT[repo].env) > 0:
+            env = " ".join(["-e %s=%s" % (k, v) for k, v in BenchRunner.CMD_URL_WAIT[repo].env.items()])
+            args = args + ' --env=\'["%s"]\'' % env.replace('"', '\\\"').replace('\'', '\'"\'"\'')
+        for a,b in BenchRunner.CMD_URL_WAIT[repo].mount:
+            a = os.path.join(os.path.dirname(os.path.abspath(__file__)), a)
+            a = tmp_copy(a)
+            args += ' --mount type=bind,src=%s,dst=%s,options=rbind' % (a,b)
+            
+        cmd = ('%s record --period=%s --file=%s/%s.txt %s %s %s/%s:latest' %
+            (optimize_bin, period, accessed_files_dir, repo , entry, args, target, repo))
+        return cmd
+
+    def optimize(self, bench, optimize_bin, target_registry, target_repo) -> str:
+        accessed_files_dir = "accessed_files"
+        if not os.path.exists(accessed_files_dir):
+            os.mkdir(accessed_files_dir)
```

Comment:
Try `os.makedirs(accessed_files_dir, exist_ok=True)`

Reply:
Get it . Thanks.